src/nvim/api/buffer.c

```diff
@@ -702,6 +706,166 @@ ArrayOf(Integer, 2) nvim_buf_get_mark(Buffer buffer, String name, Error *err)
 
   return rv;
 }
+//
+/// Returns mark info in a range (inclusive)
+/// If there are no following marks returns empty list
+///
+/// @param buffer The buffer handle
+/// @param namespace an id returned previously from extmark_ns_create
+/// @param lower any valid mark identifier (row, col) or mark_id or -1
+/// @param upper any valid mark identifier (row, col) or mark_id or -1
+/// @param amount Maximum number of extmarks to return
+/// @param reverse Decides the search direction in the range
+/// @param[out] err Details of an error that may have occurred
+/// @return [[mark_id, row, col], ...]
+ArrayOf(Object) nvim_buf_get_marks(Buffer buffer,
+                                   Integer namespace,
+                                   Object lower,
+                                   Object upper,
+                                   Integer amount,
+                                   Boolean reverse,
+                                   Error *err)
+    FUNC_API_SINCE(1)
+{
+  Array rv = ARRAY_DICT_INIT;
+  if (!ns_initialized((uint64_t)namespace)) {
+    api_set_error(err, Validation, _("Invalid mark namespace"));
+    return rv;
+  }
+
+  if(amount == 0) {
+    api_set_error(err, Validation, _("Amount must be greater than 0"));
+    return rv;
+  }
+
+  linenr_T l_lnum;
+  colnr_T l_col;
+  bool l_set = false;
+  SET_EXTMARK_INDEX_FROM_OBJ(lower, l_lnum, l_col, l_set);
```

Comment:
this could just as well be a static function in this file, then one could just do
`if(!extmark_index_from_object(lower, &l_lnum, &l_col, &err)) { return rv; }`